scalafix-core/src/main/scala/scalafix/util/Patch.scala

```diff
@@ -30,7 +32,16 @@ abstract class ImportPatch(val importer: Importer) extends TreePatch {
 }
 
 object TreePatch {
-  case class Rename(from: Name, to: Name) extends TreePatch
+  trait RenamePatch
+  case class Rename(from: Name, to: Name) extends TreePatch with RenamePatch
+  case class RenameSymbol(from: Symbol, to: Name, normalize: Boolean = false)
```

Comment:
Actually, thinking about it, I think the current `Rename` patch (RenameReference/ThisInstance) can be replaced with a 

```scala
def rename(name: Name, to: Name): Seq[TokenPatch] = {
  val tok = name.tokens.head
  Seq(
    Remove(tok),
    AddLeft(tok, to.syntax)
  )
}

```

 helper method. There's really no reason to make it a TreePatch.